pkg/tests/mux_with_test.go

```diff
@@ -0,0 +1,295 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	testutils "github.com/pulumi/pulumi-terraform-bridge/testing/x"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"github.com/pulumi/pulumi-terraform-bridge/x/muxer"
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/pkg/v3/resource/provider"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// func todo[T any]() T {
+// 	panic("TODO")
+// }
+
+func newTFProvider() *schema.Provider {
+	resource := &schema.Resource{
+		Create: func(rd *schema.ResourceData, i interface{}) (err error) {
+			v, ok := rd.GetOk("seed")
+			if !ok {
+				return fmt.Errorf(`missing parameter "seed"`)
+			}
+			err = rd.Set("value", v.(int)*10)
+			if err != nil {
+				return
+			}
+			rd.SetId("1")
+			return
+		},
+		Schema: map[string]*schema.Schema{
+			"seed": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"value": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+
+	datasource := &schema.Resource{
+		Read: func(rd *schema.ResourceData, i interface{}) (err error) {
+			err = rd.Set("number", 10)
+			if err != nil {
+				return
+			}
+			rd.SetId("1")
+			return
+		},
+		Schema: map[string]*schema.Schema{
+			"number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{},
+		ResourcesMap: map[string]*schema.Resource{
+			"random_number": resource,
+		},
+		DataSourcesMap: map[string]*schema.Resource{
+			"random_number": datasource,
+		},
+	}
+}
+
+func newProviderServer(info tfbridge.ProviderInfo) (server pulumirpc.ResourceProviderServer, err error) {
+	ctx := context.Background()
+	schema, err := tfgen.GenerateSchema(info, diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{
+		Color: colors.Never,
+	}))
+	if err != nil {
+		return nil, err
+	}
+	data, err := json.Marshal(schema)
+	if err != nil {
+		return nil, err
+	}
+	server = tfbridge.NewProvider(ctx,
+		nil,          /* hostClient */
+		info.Name,    /* module */
+		info.Version, /* version */
+		info.P,       /* tf */
+		info,         /* info */
+		data,         /* pulumiSchema */
+	)
+	return
+}
+func TestMuxWithProvider(t *testing.T) {
```

Comment:
Empty line before func?

Reply:
@t0yv0 shall I apply `gofumpt` instead of addin an empty line?

Reply:
Ah it does that? We never updated this codebase to gofumpt but it sounds like a good idea. Ignore for now I'll talk to folks.

Reply:
@t0yv0 perhaps the bridge should then adopt the `.golangci.yml` of the `pulumi/pulumi` repo.

Reply:
Yeah great idea. Let me file it up.

Reply:
@t0yv0 don't be that excited! 😁 I tested this yesterday -> plenty of linting errors!

Reply:
Done